pytorch_lightning/trainer/trainer.py

```diff
@@ -838,6 +836,23 @@ def tune(
 
         return result
 
+    def _restore_checkpoint(self) -> None:
+        # restore modules after setup
+        if self.state.fn == TrainerFn.FITTING:
+            self.checkpoint_connector.resume_start()
+        self.checkpoint_connector.restore_datamodule()
+        self.checkpoint_connector.restore_model()
+        # restore callback states
+        self.checkpoint_connector.restore_callbacks()
+
+    def _load_checkpoint_weights(self):
+        # only one process running at this point for TPUs, as spawn isn't triggered yet
+        # todo: move this logic internally within the barrier.
+        if not self._device_type == DeviceType.TPU:
+            self.training_type_plugin.barrier()
+        rank_zero_info(f"Loading checkpoint from {self._ckpt_path}")
```

Comment:
the message should probably be more specific here that we only restore the weights. 
it's not restoring training, callbacks, etc. right?
```suggestion
        rank_zero_info(f"Loading model weights from checkpoint at {self._ckpt_path}")
```

Reply:
Yep agreed, just pulled all this portion from the previous written code, this is more informative!